lib/views/insert-image-view.coffee

```diff
@@ -169,14 +169,20 @@ class InsertImageView extends View
 
   copyImage: (file, callback) ->
     return callback() if utils.isUrl(file) || !fs.existsSync(file)
+    destFileName = path.basename(file)
 
     try
-      destFile = path.join(@siteLocalDir(), @siteImagesDir(), path.basename(file))
+      if config.get("postAssetFolder")
+        destFile = path.join(path.dirname(@editor.getPath()),
+                             templateHelper.parseFileSlug(@editor.getPath()),
+                             destFileName)
```

Comment:
Eh-oh, I haven’t noticed the PR is still open,,, after 7f7d172877d70cfe2ec142d9ee9807b27d354ccc and 0ee203c9c00d3606b8a61f2a0d30d1700fa441b3 all i was thinking about is that, oh shit I need a rewrite....I'm not aware there's a new thing called slug..

--update
I tried it and somehow it worked, but this way has a problem that it can only place file in a fixed position, if I move the post to other directory not according to the md-writer config or just rename it(not mentioning the drafts' images), the folder it creates will not change, nor will it work.